### PR TITLE
Support hdf5 cross-building if the version is greater than 1.14.4-3

### PR DESCRIPTION
**Pull request by qihao727**
Specify library name and version:  **lib/1.0**

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@qihao727 Please, update the `config.yml` too with that new version.

**Review by perseoGI — DISMISSED** [review]
LGTM

File: recipes/hdf5/all/conanfile.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def requirements(self):
             self.requires("openmpi/4.1.0")
 
     def validate(self):
-        if not can_run(self):
+        if Version(self.version) < "1.14.4-3" and not can_run(self):
```
[the remark this answers is not in the document]
Comment (qihao727) [reply recipes/hdf5/all/conanfile.py]: I've made some changes based on your suggestion, would you have a look again? Thanks.

**Review by jcar87 — APPROVED** [review]
lgtm, thanks a lot @qihao727 !!!

File: recipes/hdf5/all/conandata.yml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "1.14.4-3":
```
Comment (jcar87) [inline recipes/hdf5/all/conandata.yml:2]:
```suggestion
  "1.14.4.3":
```

Comment (jcar87) [reply recipes/hdf5/all/conandata.yml:2]: @RubenRBS

File: recipes/hdf5/config.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
 versions:
+  "1.14.4-3":
```
Comment (RubenRBS) [inline recipes/hdf5/config.yml:2]:
```suggestion
  "1.14.4.3":
```

File: recipes/hdf5/all/conanfile.py [hunk 4]
```diff
@@ -100,6 +97,11 @@ def validate(self):
         if self.settings.get_safe("compiler.cppstd"):
             check_min_cppstd(self, self._min_cppstd)
 
+    def validate_build(self):
+        if cross_building(self) and Version(self.version) < "1.14.4-3":
```
Comment (RubenRBS) [inline recipes/hdf5/all/conanfile.py:101]:
```suggestion
        if cross_building(self) and Version(self.version) < "1.14.4.3":
```

**Review by RubenRBS — APPROVED** [review]
Thanks a lot! We have updated the tag to match the expectations of Conan users, backed up by the release notes at https://github.com/HDFGroup/hdf5/releases/tag/hdf5_1.14.4.3 in the `The release tag in GitHub will be hdf5_1.14.4.3` section :)

**Review by jcar87 — APPROVED** [review]

**Pull request merged**